ts/src/header-validator/validate-json.ts

```diff
@@ -725,6 +725,102 @@ const priorityField: StructFields<Priority> = {
   priority: field('priority', int64, 0n),
 }
 
+function isDebugTypeSupported(ctx: RegistrationContext, s: string) {
+  if (ctx instanceof SourceContext) {
+    return constants.sourceAggregateDebugTypes.includes(s)
+  } else {
+    return constants.triggerAggregateDebugTypes.includes(s)
+  }
```

Comment:
Instead of doing an `instanceof` check here, can we change the context type to itself have a field for the debug types?